server/controllers/profile.js

```diff
@@ -114,4 +116,39 @@ exports.findAllProfiles = asyncHandler(async (req, res, next) => {
       }
 
     res.status(200).json({ profiles: profiles });
-})
\ No newline at end of file
+})
+
+const deletePhoto = async (user) => {
+  await cloudinary.uploader.destroy(user.profilePhoto.publicId);
+  user.profilePhoto = null;
+};
+
+exports.uploadProfilePhoto = asyncHandler(async (req, res, next) => {
+  const image = req.file;
+  const user = await Profile.findById(req.user.id);
```

Comment:
Always validate the request first so you can return a bad request status to the client if the required fields are missing.